src/main/java/ch/njol/skript/lang/SkriptEvent.java

```diff
@@ -241,4 +284,41 @@ public static SkriptEvent parse(String expr, SectionNode sectionNode, @Nullable
 		return (SkriptEvent) Structure.parse(expr, sectionNode, defaultError, Skript.getEvents().iterator());
 	}
 
+	/**
+	 * The listening behavior of a Skript event. This determines whether the event should be run for cancelled events, uncancelled events, or both.
+	 */
+	public enum ListeningBehavior {
+		/**
+		 * This Skript event should be run for any uncancelled event.
+		 */
+		UNCANCELLED,
+		/**
+		 * This Skript event should be run for any cancelled event.
+		 */
+		CANCELLED,
+		/**
+		 * This Skript event should be run for any event, cancelled or uncancelled.
+		 */
```

Comment:
I know I know I'm no grammar person but isn't correct format "This Skript event should be ran for" that or remove "be" as "This Skript event should run for any ..." seems correct to me.

"be run" just isn't correct afaik

Didn't suggest since this is apart of all javadocs for this enum

Reply:
"should be run for" is correct, as far as I know, since it's not referring to the past.
I do agree that "should run" just seems simpler and better though.